File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    result = []
    for a1, a2 in zip(arr1, arr2):
        v = bin(a1 | a2)[2:].zfill(n).replace("1", "#").replace("0", " ")
        result.append(v)
    return result
        
        

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] result = new String[n];
        
        for (int i = 0; i < n; i++) {
            String v = Integer.toBinaryString(arr1[i] | arr2[i]);
            StringBuilder binary = new StringBuilder(v);
            
            while (binary.length() < n) {
                binary.insert(0, "0");
            }
            
            binary = new StringBuilder(binary.toString().replaceAll("1", "#"));
            binary = new StringBuilder(binary.toString().replaceAll("0", " "));
            
            result[i] = binary.toString();
        }
        
        return result;
    }
}